core/jobs/batch_jobs/exp_version_history_computation_job.py

```diff
@@ -0,0 +1,1230 @@
+                            vh_model.state_version_history.items()
+                        )
+                    }
+                    actual_metadata_vh = exp_domain.MetadataVersionHistory(
+                        vh_model.metadata_last_edited_version_number,
+                        vh_model.metadata_last_edited_committer_id
+                    )
+                    if not self.compare_version_histories(
+                        expected_state_vh, expected_metadata_vh,
+                        actual_state_vh, actual_metadata_vh
+                    ):
+                        verified = False
+                        break
+                else:
+                    expected_state_vh = (
+                        exp_services.update_states_version_history(
+                            copy.deepcopy(verified_state_vh[-1]),
+                            change_list, prev_exp.states,
+                            curr_exp.states, version, committer_id
+                        )
+                    )
+                    expected_metadata_vh = (
+                        exp_services.update_metadata_version_history(
+                            copy.deepcopy(verified_metadata_vh[-1]),
+                            change_list, prev_exp.get_metadata(),
+                            curr_exp.get_metadata(), version, committer_id
+                        )
+                    )
+                    actual_state_vh = {
+                        state_name: state_domain.StateVersionHistory.from_dict(
+                            state_vh_dict
+                        )
+                        for state_name, state_vh_dict in (
+                            vh_model.state_version_history.items()
+                        )
+                    }
+                    actual_metadata_vh = exp_domain.MetadataVersionHistory(
+                        vh_model.metadata_last_edited_version_number,
+                        vh_model.metadata_last_edited_committer_id
+                    )
+                    if not self.compare_version_histories(
+                        expected_state_vh, expected_metadata_vh,
+                        actual_state_vh, actual_metadata_vh
+                    ):
+                        verified = False
+                        break
+                verified_state_vh.append(expected_state_vh)
+                verified_metadata_vh.append(expected_metadata_vh)
+
+        return (exp_id, verified)
+
+    def run(self) -> beam.PCollection[job_run_result.JobRunResult]:
+        all_explorations_vlatest = (
+            self.pipeline
+            | 'Get all ExplorationModels' >> ndb_io.GetModels(
+                exp_models.ExplorationModel.get_all(include_deleted=False)
+            )
+            | 'Create key-value pairs with id and exp models' >>
+                beam.Map(lambda model: (model.id, model))
+        )
+
+        all_explorations_v1 = (
+            all_explorations_vlatest
```

Comment:
```suggestion
            self.pipeline
```
ditto elsewhere where `GetModels`  is used.

Reply:
Done.